lib/modules/keyboardNav.js

```diff
@@ -890,42 +893,85 @@ const getActiveCommandOptions = _.once(() =>
 const _commandLookup = _.once(() => {
 	const lookup = {};
 	for (const option of [...getActiveCommandOptions(), ...getCommentsLinkKeys()]) {
+		if (!option.value) continue;
 		const hash = hashKeyArray(option.value);
 		if (!lookup[hash]) {
-			lookup[hash] = $.Callbacks();
+			lookup[hash] = [];
 		}
 
-		if (option.goMode) lookup[hash].add(() => { handleGoModeCommand((option.callback: any)); });
-		else lookup[hash].add(() => { if (!goModeActive) option.callback(); });
+		lookup[hash].push(option);
 	}
 	return lookup;
 });
 
-function getCommandsForKeyEvent(keyEvent) {
-	const hash = hashKeyEvent(keyEvent);
-	return _commandLookup()[hash];
+function promptUniqueShortcutSelection(options) {
+	const shortcut = niceKeyCode(options[0].value);
+
+	const optionButtons = options.map(option => {
+		const button = $(`<button style="width: 100%; margin-top: 2px;">${i18n(option.description)}</button>`).get(0);
+		button.addEventListener('click', () => { optionSelected(option); });
+		return button;
+	});
+
+	const question = i18n('keyboardNavAmbiguousShortcutPrompt', i18n(module.moduleName), `<i>${shortcut}</i>`, '<br><br>', `<i>${shortcut}</i>`);
+	const followUp = i18n('keyboardNavAmbiguousShortcutPromptFollowUp', SettingsNavigation.makeUrlHashLink(module.moduleID, undefined, i18n(module.moduleName)));
+
+	const message = $(`<div>${question}</div>`)
+		.append(optionButtons)
+		.append(`<br><br><small>(${followUp})</small>`)
+		.get(0);
+
+	const notification = Notifications.showNotification({
+		header: 'Ambiguous shortcut',
+		message,
+		noDisable: true,
+		closeDelay: Infinity,
+	});
+
+	async function optionSelected(option) {
+		for (const v of options.filter(v => v !== option)) {
+			const [name] = Object.entries(module.options).find(([, value]) => value === v) || [];
+			if (name) Options.set(module, name, null);
+		}
+
+		option.callback();
+		(await notification).close();
+	}
 }
 
 function handleKeyPress(e) {
 	if (EasterEgg.konamiActive()) return false;
 
-	const callbacks = getCommandsForKeyEvent(e);
-
-	if (!callbacks) return false;
-
 	// Allow navigation on other elements when input has no (apparent) default behavior
 	if (['A', 'BUTTON'].includes(document.activeElement.tagName)) {
 		const hasDefaultBehavior = [/* tab */9, /* enter */13, /* space */32].includes(e.which);
-
-		if (!hasDefaultBehavior) {
-			document.activeElement.blur();
-		}
+		if (hasDefaultBehavior) return false;
+	} else if (document.activeElement.tagName !== 'BODY') {
+		return false;
 	}
 
-	if (document.activeElement.tagName === 'BODY') {
-		callbacks.fire(e);
-		return true;
+	const hash = hashKeyEvent(e);
+	const options = (_commandLookup()[hash] || [])
+		.filter(v => {
+			if (v.goMode) return module.options.useGoMode.value === goModeActive;
+			else return !goModeActive;
+		});
+
+	if (!options.length) {
+		return false;
+	} else if (options.length > 1 && options.some(v => v.goMode)) {
```

Comment:
The reason why this is limited to `goMode` is that there exist shortcuts coexist on the same key, e.g. `saveLink` and `saveComment`.